Tasks/ServiceFabricUpdateManifests/Update-ApplicationVersions.psm1

```diff
@@ -57,20 +57,45 @@
 
                 Write-Host (Get-VstsLocString -Key SearchingApplicationType -ArgumentList $appTypeName)
 
-                $oldAppPackagePath = Join-Path $oldDropLocation $newAppPackagePath.SubString((Get-VstsTaskVariable -Name Build.SourcesDirectory -Require).Length + 1)
-                $oldAppManifestPath = Join-Path $oldAppPackagePath $appManifestName
-                if (Test-Path $oldAppManifestPath)
+                # Try and find the old app package path by finding the largest substring of the path that exists in the artifact path
+                $relativePath = $newAppPackagePath
```

Comment:
Shouldn't we start with Build.SourcesDirectory directory rather than root?

Reply:
Generally this would actually be under Build.StagingDirectory, but there is no guarantee of that.

Reply:
got it